### DISCUSSION TITLE
Extreme HTTP Performance Tuning: 1.2M API req/s

### DISCUSSION THREAD

**Discussion by uasan** [uNetworking/uWebSockets.js#531] [General]
I think you will be interested
https://talawah.io/blog/extreme-http-performance-tuning-one-point-two-million/

**Comment by hst-m** [comment]
I saw that yesterday on hacker news, still on front page https://news.ycombinator.com/item?id=27226382

**Comment by dalisoft** [comment]
Thanks @uasan for article, but how it's related to this project? This project works fast itself and there one more thing: HTTP Server itself not slow, slow is it's what inside - Database query, logic, calculation and etc.
Really big thank you for article @uasan

**Reply by uasan** [reply]
This article shows you how you can improve performance by tweaking the OS without changing your application code.

**Reply by dalisoft** [reply]
I am tried and can say more than of half optimizations cannot be applied to uWebSockets (C++ version) and either does not need as this library fast already, i didn't see direct benchmarking, so i don't think it's reality.
The real bottleneck is logic, parser, converter, queries and other many factors. These ~1.2M API req/s is raw text, through it's looks like great but it's WRONG as it's at cost of security was gained which on Enterprise servers isn't option, never.
Maybe i don't know enough or wrong, but i'm happy with results of uWebSockets.js, it's battle tested on most popular production tools like Trello.
If there any optimizations available, sure, @alexhultman does and he knows better, at all he made very high performance yet available server framework for Node.js.

I am not against these optimizations, but some of newbies applies these optimizations on their first work says it's nice, but their bussiness keeps under security issues.

If there any errors in text, sorry, as english isn't my native language.
